src/backend/ggml.rs

```diff
@@ -435,7 +451,127 @@ pub(crate) async fn rag_query_handler(mut req: Request<Body>) -> Response<Body>
+                    info!(target: "stdout", "No points retrieved, enabling web search.");
+                    web_search_allowed = true;
+                }
+            }
+        }
+    }
+
+    #[cfg(feature = "search")]
+    if web_search_allowed {
+        let search_arguments = match crate::SEARCH_ARGUMENTS.get() {
+            Some(sc) => sc,
+            None => {
+                let err_msg = "Failed to obtain SEARCH_ARGUMENTS. Was it set?".to_string();
+                error!(target: "stdout", "{}", &err_msg);
+
+                return error::internal_server_error(err_msg);
             }
+        };
+
+        let endpoint: hyper::Uri = match search_arguments.query_server_url.parse() {
+            Ok(uri) => uri,
+            Err(e) => {
+                let err_msg = format!("LlamaEdge Query server URL could not be parsed: {}", e);
+                error!(target: "stdout", "{}", &err_msg);
+
+                return error::internal_server_error(err_msg);
+            }
+        };
+
+        let summary_endpoint = match hyper::Uri::builder()
+            .scheme(endpoint.scheme().unwrap().to_string().as_str())
+            .authority(endpoint.authority().unwrap().to_string().as_str())
+            .path_and_query("/query/summarize")
+            .build()
+        {
+            Ok(se) => se,
+            Err(_) => {
+                let err_msg = "couldn't build summary_endpoint from query_server_url".to_string();
+                error!(target: "stdout", "{}", &err_msg);
+
+                return error::internal_server_error(err_msg);
+            }
+        };
+
+        //perform query, extract summary, add to
+        let req = match Request::builder()
+            .method(Method::POST)
+            .uri(summary_endpoint)
+            .header("content-type", "application/json")
+            .body(Body::from(
+                serde_json::json!({
+                    "search_config" : {
+                    "api_key": search_arguments.api_key,
+                    },
+                    "backend": search_arguments.search_backend,
+                    "query": query,
+                })
+                .to_string(),
+            )) {
+            Ok(request) => request,
+            Err(_) => {
+                let err_msg = "failed to build request to LLamaEdge query server.".to_string();
+                error!(target: "stdout", "{}", &err_msg);
+                return error::internal_server_error(err_msg);
+            }
+        };
+
+        info!(target: "stdout", "Querying the LlamaEdge query server.");
+
+        let client = hyper::client::Client::new();
+        let res = match client.request(req).await {
+            Ok(response) => response,
+            Err(e) => {
+                let err_msg = format!("couldn't make request to LlamaEdge query server: {}", e);
+                error!(target: "stdout", "{}", &err_msg);
+
+                return error::internal_server_error(err_msg);
+            }
+        };
```

Comment:
It would be necessary to check if the query server is available or not. If it's not available, should guarantee that the rag workflow does not `crash` or `stop` here; instead, it should go into the next step with the original chat messages. Thanks!

Reply:
I've made it `warn!` instead of `error!`ing. Also, no error is returned. Please recheck.